### DISCUSSION TITLE
Mega2560 and Mux Shield 2

### DISCUSSION THREAD

**Discussion by siejeje** [shanteacontrols/OpenDeck#296] [Q&A] [answered]
Hi!
I flashed Mega2560 board according to instructions and it works flawlessly with the standard firmware version without Mux Shield (mega2560.hex file). It is recognized by the system right after connecting and web configurator works like a charm. Of course I flashed 16u2 MCU first.

However when I flash it with mega2560_muxsh2.hex writing completes successfully but after connecting the board to the computer directly without programmer the board is not recognized by the system (not appearing as usb device neither in Windows nor in Macos). OpenDeck configurator says "No OpenDeck board found. Please connect the board in order to use the interface." Regardless if I connect it with Mux Shield 2 installed on the top of arduino or not.

Is there anything specific I need to do to make it work with Mux Shield apart from loading dedicated firmware?

I'm flashing on M1 Macbook using avrdude and arduino uno as a programmer.
Thanks

EDIT: I noticed that "L" led on the Mega board is constantly glowing after flashing mega2560_muxsh2.hex variant while it is off with standard firmware. Not sure if this means anything.
 
Here is my board with Mux Shield installed:
![WhatsApp Image 2024-06-02 at 22 47 26](https://github.com/shanteacontrols/OpenDeck/assets/10771632/7889e6bf-bb9f-474f-af40-2ef12349e9bf)

**Comment by siejeje** [comment]
And here is the log from flashing, if it helps in any way:

tomek@MacBook-Air-Tomek ~ % "/Users/tomek/Library/Arduino15/packages/arduino/tools/avrdude/6.3.0-arduino17/bin/avrdude" "-C/Users/tomek/Library/Arduino15/packages/arduino/tools/avrdude/6.3.0-arduino17/etc/avrdude.conf" -v -V -patmega2560 -cstk500v1 -P/dev/cu.usbmodem1201 -b19200 "-Uflash:w:/Users/tomek/Downloads/mega2560_muxsh2.hex:i"

avrdude: Version 6.3-20190619
         Copyright (c) 2000-2005 Brian Dean, http://www.bdmicro.com/
         Copyright (c) 2007-2014 Joerg Wunsch

         System wide configuration file is "/Users/tomek/Library/Arduino15/packages/arduino/tools/avrdude/6.3.0-arduino17/etc/avrdude.conf"
         User configuration file is "/Users/tomek/.avrduderc"
         User configuration file does not exist or is not a regular file, skipping

         Using Port                    : /dev/cu.usbmodem1201
         Using Programmer              : stk500v1
         Overriding Baud Rate          : 19200
         AVR Part                      : ATmega2560
         Chip Erase delay              : 9000 us
         PAGEL                         : PD7
         BS2                           : PA0
         RESET disposition             : dedicated
         RETRY pulse                   : SCK
         serial program mode           : yes
         parallel program mode         : yes
         Timeout                       : 200
         StabDelay                     : 100
         CmdexeDelay                   : 25
         SyncLoops                     : 32
         ByteDelay                     : 0
         PollIndex                     : 3
         PollValue                     : 0x53
         Memory Detail                 :

                                  Block Poll               Page                       Polled
           Memory Type Mode Delay Size  Indx Paged  Size   Size #Pages MinW  MaxW   ReadBack
           ----------- ---- ----- ----- ---- ------ ------ ---- ------ ----- ----- ---------
           eeprom        65    10     8    0 no       4096    8      0  9000  9000 0x00 0x00
           flash         65    10   256    0 yes    262144  256   1024  4500  4500 0x00 0x00
           lfuse          0     0     0    0 no          1    0      0  9000  9000 0x00 0x00
           hfuse          0     0     0    0 no          1    0      0  9000  9000 0x00 0x00
           efuse          0     0     0    0 no          1    0      0  9000  9000 0x00 0x00
           lock           0     0     0    0 no          1    0      0  9000  9000 0x00 0x00
           calibration    0     0     0    0 no          1    0      0     0     0 0x00 0x00
           signature      0     0     0    0 no          3    0      0     0     0 0x00 0x00

         Programmer Type : STK500
         Description     : Atmel STK500 Version 1.x firmware
         Hardware Version: 2
         Firmware Version: 1.18
         Topcard         : Unknown
         Vtarget         : 0.0 V
         Varef           : 0.0 V
         Oscillator      : Off
         SCK period      : 0.1 us

avrdude: AVR device initialized and ready to accept instructions

Reading | ################################################## | 100% 0.03s

avrdude: Device signature = 0x1e9801 (probably m2560)
avrdude: safemode: lfuse reads as FF
avrdude: safemode: hfuse reads as D8
avrdude: safemode: efuse reads as FD
avrdude: NOTE: "flash" memory has been specified, an erase cycle will be performed
         To disable this feature, specify the -D option.
avrdude: erasing chip
avrdude: reading input file "/Users/tomek/Downloads/mega2560_muxsh2.hex"
avrdude: writing flash (77556 bytes):

Writing | ################################################## | 100% 73.23s

avrdude: 77556 bytes of flash written

avrdude: safemode: lfuse reads as FF
avrdude: safemode: hfuse reads as D8
avrdude: safemode: efuse reads as FD
avrdude: safemode: Fuses OK (E:FD, H:D8, L:FF)

avrdude done.  Thank you.

**Comment by siejeje** [comment]
Today I set up the environment on Ubuntu 22.04, including installing Docker and cloning OpenDeck repository as I thought maybe there is something wrong with flashing on Mac.
I was able to successfully build and flash the firmware but unfortunately the result is exactly the same as on Mac. mega2560 firmware works perfectly but mega2560_muxsh2 won't boot at all.
At this point I've run out of ideas, @paradajz you are my last resort. Appreciate any help on this.

**Comment by paradajz** [comment, marked as answer]
It's most likely that the configuration for the variant with Mux Shield v2 uses too much RAM and hence crashes. Try removing some of the stuff you don't use from the configuration - touchscreen, I2C display, DIN MIDI etc.

**Reply by paradajz** [reply]
I've actually just disabled touchscreen on that config by default

**Comment by siejeje** [comment]
I removed dinMIDI and i2c from the config file and boom! it worked perfectly :)
Thank you very much for your support and for your great project.
